tests/components/humidifier/test_device_trigger.py

```diff
@@ -0,0 +1,353 @@
+    # Fake that the humidity is changing
+    hass.states.async_set("humidifier.entity", STATE_ON, {const.ATTR_HUMIDITY: 37})
+    await hass.async_block_till_done()
+    assert len(calls) == 2
+    assert calls[1].data["some"] == "target_humidity_changed_above"
+
+    # Wait 6 minutes
+    async_fire_time_changed(hass, dt_util.utcnow() + datetime.timedelta(minutes=6))
+    await hass.async_block_till_done()
+    assert len(calls) == 3
+    assert calls[2].data["some"] == "target_humidity_changed_above_for"
+
+    # Fake turn off
+    hass.states.async_set("humidifier.entity", STATE_OFF, {const.ATTR_HUMIDITY: 37})
+    await hass.async_block_till_done()
+    assert len(calls) == 4
+    assert (
+        calls[3].data["some"] == "turn_off device - humidifier.entity - on - off - None"
+    )
+
+    # Fake turn on
+    hass.states.async_set("humidifier.entity", STATE_ON, {const.ATTR_HUMIDITY: 37})
+    await hass.async_block_till_done()
+    assert len(calls) == 5
+    assert (
+        calls[4].data["some"] == "turn_on device - humidifier.entity - off - on - None"
+    )
+
+
+async def test_invalid_config(hass, calls):
+    """Test for turn_on and turn_off triggers firing."""
+    hass.states.async_set(
+        "humidifier.entity",
+        STATE_ON,
+        {
+            const.ATTR_HUMIDITY: 23,
+            const.ATTR_MODE: "home",
+            const.ATTR_AVAILABLE_MODES: ["home", "away"],
+            ATTR_SUPPORTED_FEATURES: 1,
+        },
+    )
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {
+                        "platform": "device",
+                        "domain": DOMAIN,
+                        "device_id": "",
+                        "entity_id": "humidifier.entity",
+                        "type": "target_humidity_changed",
+                        "below": 20,
+                        "invalid": "invalid",
+                    },
+                    "action": {
+                        "service": "test.automation",
+                        "data_template": {"some": "target_humidity_changed"},
+                    },
+                },
+            ]
+        },
+    )
+
+    # Fake that the humidity is changing
+    hass.states.async_set("humidifier.entity", STATE_ON, {const.ATTR_HUMIDITY: 7})
+    await hass.async_block_till_done()
+    # Should not trigger for invalid config
+    assert len(calls) == 0
+
+
+async def test_get_trigger_capabilities_on(hass):
+    """Test we get the expected capabilities from a humidifier trigger."""
+    capabilities = await device_trigger.async_get_trigger_capabilities(
+        hass,
+        {
+            "platform": "device",
+            "domain": "humidifier",
+            "type": "turned_on",
+            "entity_id": "humidifier.upstairs",
+            "above": "23",
+        },
+    )
+
+    assert capabilities and "extra_fields" in capabilities
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)